tests/unit/worker/file_util_test.py

```diff
@@ -171,7 +173,8 @@ def test_has_files(self):
         self.assertNotIn('b.txt', output_dir_entries)
         self.assertTrue(os.path.exists(os.path.join(output_dir, 'dir1', 'f1')))
         self.assertFalse(os.path.exists(os.path.join(output_dir, 'dir1', 'f2')))
-        self.assertTrue(os.path.islink(os.path.join(output_dir, 'a-symlink.txt')))
+        if not self.skip_symlinks:
+            self.assertTrue(os.path.islink(os.path.join(output_dir, 'a-symlink.txt')))
```

Comment:
Technically, this is a breaking change -- when uploading a .zip file, symlinks are no longer preserved -- but I don't think this is a feature that our users rely on. We used to never preserve symlinks in .zip file uploads; we only started doing so after https://github.com/codalab/codalab-worksheets/pull/2952, and we never documented that fact.

So I think it should be OK to remove support for symlinks now. (We can't preserve symlinks anymore since we're using Python's ZipFile now, which doesn't have support for symlinks -- unless we make further modifications to it).

Reply:
what's wrong with making further modifications to `ZipFile`? I think we should be careful here

Reply:
symlinks themselves aren't part of the .zip standard. They're just sometimes added as part of extra metadata fields.

In fact, normally if you create a .zip file, symlinks aren't preserved by default unless you actually run it with the `zip --symlinks` option (see https://serverfault.com/questions/265675/how-can-i-zip-compress-a-symlink).

The majority of cases in which we are handling .zip files anyway will be through uploads that happen through the web interface -- and in this case, the zip file is created without the `--symlinks` option.

So what this really breaks is the use case in which someone 1) created a zip file with symlinks in it with `zip --symlinks` and then 2) uploads it to codalab and it gets unpacked. It seems like this is an edge-enough use case (which would only have worked since October 2020) that we can just remove it.